python/noxfile.py

```diff
@@ -149,6 +153,10 @@ def docs(session: nox.Session) -> None:
 @nox.session(name="docs-build", python=python_versions[0])
 def docs_build(session: nox.Session) -> None:
     """Build the docs."""
+    if not shutil.which("quarto"):
+        session.error("Unable to execute `quarto`. Is it installed and in your path?\n"
+                      "See https://quarto.org/docs/get-started/")
+
```

Comment:
i wish nox had a way to call other nox commands before.  like dependency running this one first. so we could avoid duplication like this.

it has a way to run other nox commands afterwards, but not before.

Reply:
Yep. I like that it lets us write Python to check things -- makes it easy for cases like this. I like that it makes it easy to run all python versions using virtualenvs. I'm watching https://github.com/wntrblm/nox/issues/453 for dependency support (would be awesome to only build the maturin extension once, for instance).

It looks like https://github.com/rmorshea/noxopt#noxopt may be worth keeping an eye on too -- it would let us register a setup method for a prefix of the commands.